src/form/Select/Select.tsx

```diff
@@ -92,7 +92,7 @@ export interface SelectProps {
   /**
    * Whether you can filter the select options.
    */
-  filterable?: boolean;
+  filterable?: boolean | 'async';
```

Comment:
i don't love the mixing of a boolean type w/ something else but i guess without breaking changes, this seems to be the best option unless we add a filter type prop